src/openms_gui/source/VISUAL/DIALOGS/FLASHDeconvTabWidget.cpp

```diff
@@ -0,0 +1,436 @@
+    {
+      // create a default INI of FLASHDeconv
+      String tmp_file = File::getTemporaryFile();
+      if (ep_.run(this, getFLASHDeconvExe().toQString(), QStringList() << "-write_ini" << tmp_file.toQString(), "", true) != ExternalProcess::RETURNSTATE::SUCCESS)
+      {
+        exit(1);
+      }
+      ParamXMLFile().load(tmp_file, flashdeconv_param_);
+      flashdeconv_param_ = flashdeconv_param_.copy("FLASHDeconv:1:", true);
+
+      // parameters to show in default mode : flashdeconv_param_wizard_
+      flashdeconv_param_.remove("log");
+      flashdeconv_param_.remove("no_progress");
+      flashdeconv_param_.remove("debug");
+      flashdeconv_param_.remove("in");
+      flashdeconv_param_.remove("out");
+
+      // parameters for different output format
+      StringList out_params = {"out_spec", "out_mzml", "out_promex", "out_topFD", "out_topFD_feature"};
+      for (const auto& name : out_params) flashdeconv_param_outputs_.setValue(name, ""); // create a dummy param, just so we can use ::copySubset
+      flashdeconv_param_outputs_ = flashdeconv_param_.copySubset(flashdeconv_param_outputs_);
+
+      // remove output format params from global parameter set
+      for (const auto& name : out_params) flashdeconv_param_.remove(name);
+
+      ui->list_editor->load(flashdeconv_param_);
+    }
+
+    QString FLASHDeconvTabWidget::getCurrentOutDir_() const
+    {
+      QString out_dir(ui->out_dir->dirNameValid() ?
+        ui->out_dir->getDirectory() :
+        getFDDefaultOutDir());
+      return out_dir;
+    }
+
+    void FLASHDeconvTabWidget::writeLog_(const QString& text, const QColor& color, bool new_section)
+    {
+      QColor tc = ui->log_text->textColor();
+      if (new_section)
+      {
+        ui->log_text->setTextColor(Qt::darkBlue);
+        ui->log_text->append(QString(10, '#').append(QDateTime::currentDateTime().toString("yyyy-MM-dd hh:mm:ss")).append(QString(10, '#')).append("\n"));
+        ui->log_text->setTextColor(tc);
+      }
+
+      ui->log_text->setTextColor(color);
+      ui->log_text->append(text);
+      ui->log_text->setTextColor(tc); // restore old color
+    }
+
+    void FLASHDeconvTabWidget::writeLog_(const String& text, const QColor& color, bool new_section)
+    {
+      writeLog_(text.toQString(), color, new_section);
+    }
+    
+    bool FLASHDeconvTabWidget::checkFDInputReady_()
+    {
+      if (ui->input_mzMLs->getFilenames().empty())
+      {
+        QMessageBox::critical(this, "Error", "Input mzML file(s) are missing! Please provide at least one!");
+        return false;
+      }
+
+      return true;
+    }
+
+    void FLASHDeconvTabWidget::broadcastNewCWD_(const QString& new_cwd)
+    {
+      // RAII to avoid infinite loop (setCWD signals updatedCWD which is connected to slot broadcastNewCWD_)
+      QSignalBlocker blocker1(ui->input_mzMLs);
+      ui->input_mzMLs->setCWD(new_cwd);
+    }
+
+    /// custom arguments to allow for looping calls
+    struct Args
```

Comment:
a lot of code duplication from the original wizard.
Please move it into a separate class and just call it from both wizards

Reply:
were would you put it?
Some WizardHelper class?

Reply:
yep, sounds good